mlrun/runtimes/nuclio/application/application.py

```diff
@@ -488,6 +491,39 @@ def invoke(
             **http_client_kwargs,
         )
 
+    @classmethod
+    def deploy_reverse_proxy_image(cls):
+        """
+        Build the reverse proxy image and save it.
+        The reverse proxy image is used to route requests to the application sidecar.
+        This is useful when you want to decrease build time by building the application image only once.
+
+        :param use_cache:   Use the cache when building the image
+        """
+        # create a function that includes only the reverse proxy, without the application
+
+        reverse_proxy_func = mlrun.run.new_function(
+            name="reverse-proxy-temp", kind="remote"
+        )
+        # default max replicas is 4, we only need one replica for the reverse proxy
+        reverse_proxy_func.spec.max_replicas = 1
+
+        # clear out the base image, so it won't be mlrun/mlrun
+        reverse_proxy_func.set_config("spec.build.baseImage", None)
+        reverse_proxy_func.spec.image = ""
+        reverse_proxy_func.spec.build.base_image = ""
```

Comment:
why is this needed as well as `skip_base_image_enrichment` ?
a more fitting comment would be "the reverse proxy image should not be based on another image" (or use whatever magic nuclio does

Reply:
Because of this: https://github.com/mlrun/mlrun/blob/2bd77151e1c1a1051ddfd141cecc191cc79dbcc2/server/api/crud/runtimes/nuclio/function.py#L686-L690

Updated the comment

Reply:
yes but its just sanity right? because these should be empty anyway

Reply:
Generally the should be empty but nothing stops the user from setting them.
So yes, sanity.